clang/lib/Sema/SemaExpr.cpp

```diff
@@ -7274,12 +7274,19 @@ Sema::BuildCompoundLiteralExpr(SourceLocation LParenLoc, TypeSourceInfo *TInfo,
       // init a VLA in C++ in all cases (such as with non-trivial constructors).
       // FIXME: should we allow this construct in C++ when it makes sense to do
       // so?
-      std::optional<unsigned> NumInits;
-      if (const auto *ILE = dyn_cast<InitListExpr>(LiteralExpr))
-        NumInits = ILE->getNumInits();
-      if ((LangOpts.CPlusPlus || NumInits.value_or(0)) &&
-          !tryToFixVariablyModifiedVarType(TInfo, literalType, LParenLoc,
-                                           diag::err_variable_object_no_init))
+      //
+      // But: C99-C23 6.5.2.5 Compound literals constraint 1: The type name
+      // shall specify an object type or an array of unknown size, but not a
+      // variable length array type. This seems odd, as it allows int a[size] =
+      // {}; but forbids int a[size] = (int[size]){}; As this is what the
```

Comment:
technically speaking `int a[size] = (int[size]){};` is disallowed in **ALL** C standards. A better example would be:
```
int * a = (int[size]){};
```

Reply:
Ah yeah, true, you can’t do that with non-variable-length arrays either. We can update that comment as an NFC fix.

Reply:
Alright, done.

Reply:
Sorry, I should have been more clear, I didn't mean it would be allowed in another C standard, just that it's unexpected from a language point of view. But it's not that important anyway, so I hope the new version works better :).